src/hotspot/share/opto/superword.cpp

```diff
@@ -1630,36 +1621,88 @@ uint SuperWord::max_implemented_size(const Node_List* pack) {
   }
 }
 
-// Java API for Long.bitCount/numberOfLeadingZeros/numberOfTrailingZeros
-// returns int type, but Vector API for them returns long type. To unify
-// the implementation in backend, superword splits the vector implementation
-// for Java API into an execution node with long type plus another node
-// converting long to int.
-bool SuperWord::requires_long_to_int_conversion(int opc) {
-  switch(opc) {
-    case Op_PopCountL:
-    case Op_CountLeadingZerosL:
-    case Op_CountTrailingZerosL:
-      return true;
-    default:
-      return false;
+// If the j-th input for all nodes in the pack is the same unique input: return it, else nullptr.
+Node* PackSet::isa_unique_input_or_null(const Node_List* pack, int j) const {
```

Comment:
Two things here:
1. You do not seem to be using the actual unique input at the call sites and only care about whether there is a unique input or not (all checks are against `nullptr`).
2.  I'm not sure if "unique" could be misleading as in "only once" in the entire pack. 

To address both, I suggest to change this to:
`bool PackSet::same_input_at_index()`

What do you think?

Reply:
> You do not seem to be using the actual unique input at the call sites and only care about whether there is a unique input or not (all checks are against nullptr).

I do not need it now, but will after a further refactoring. So I'd like to have it already available to keep the noise in the next refactoring down.

> I'm not sure if "unique" could be misleading as in "only once" in the entire pack.

Ok, I could do that. That would be closer to the old name `same_input`.

Reply:
`same_inputs_at_index_or_null()`